src/main/java/de/tum/in/www1/artemis/service/FileUploadSubmissionService.java

```diff
@@ -106,28 +115,47 @@ public FileUploadSubmission handleFileUploadSubmission(FileUploadSubmission file
+            submissions = participations.stream().filter(studentParticipation -> studentParticipation.findLatestSubmission().isPresent())
+                    .map(StudentParticipation::findLatestSubmission).map(Optional::get).collect(toList());
+        }
+        else {
+            submissions = this.submissionRepository.findAllByParticipationExerciseIdAndResultAssessor(exerciseId, tutor);
+        }
         return submissions.stream().map(submission -> {
             submission.getResult().setSubmission(null);
             return (FileUploadSubmission) submission;
         }).collect(Collectors.toList());
     }
 
     /**
-     * Given an exercise id, find a random file upload submission for that exercise which still doesn't have any manual result. No manual result means that no user has started an
-     * assessment for the corresponding submission yet.
+     * Given an exercise id, find a random file upload submission for that exercise which still doesn't have any manual result.
+     * No manual result means that no user has started an assessment for the corresponding submission yet.
+     * For exam exercises we should also remove the test run participations as these should not be graded by the tutors.
      *
      * @param fileUploadExercise the exercise for which we want to retrieve a submission without manual result
+     * @param removeTestRunParticipations flag to determine if test runs should be removed. This should be set to true for exam exercises
      * @return a fileUploadSubmission without any manual result or an empty Optional if no submission without manual result could be found
      */
     @Transactional(readOnly = true)
-    public Optional<FileUploadSubmission> getFileUploadSubmissionWithoutManualResult(FileUploadExercise fileUploadExercise) {
+    public Optional<FileUploadSubmission> getRandomFileUploadSubmissionEligibleForNewAssessment(FileUploadExercise fileUploadExercise, boolean removeTestRunParticipations) {
         Random random = new Random();
-        var participations = participationService.findByExerciseIdWithLatestSubmissionWithoutManualResults(fileUploadExercise.getId());
-        var submissionsWithoutResult = participations.stream().map(StudentParticipation::findLatestSubmission).filter(Optional::isPresent).map(Optional::get)
+        List<StudentParticipation> participations;
+        if (removeTestRunParticipations) {
+            participations = participationService.findByExerciseIdWithLatestSubmissionWithoutManualResultsAndNoTestRun(fileUploadExercise.getId());
+        }
+        else {
+            participations = participationService.findByExerciseIdWithLatestSubmissionWithoutManualResults(fileUploadExercise.getId());
+        }
+
+        List<FileUploadSubmission> submissionsWithoutResult = participations.stream().filter(studentParticipation -> !studentParticipation.isTestRunParticipation())
```

Comment:
why do we filter TestRunParticipations here when we have the if else above?

Reply:
This was left over code, i removed it. Nice catch :)